tools/wpt/tests/test_update_expectations.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+
+import pytest
+
+from tools.wpt import wpt
+from tools.wptrunner.wptrunner import manifestexpected
+from localpaths import repo_root
+
+@pytest.fixture
+def metadata_file(tmp_path):
+    created_files = []
+
+    def create_metadata(test_id, subtest_name, product, status="OK", subtest_status="PASS", channel="nightly"):
+        run_info = {
+            "os": "linux",
+            "processor": "x86_64",
+            "version": "Ubuntu 20.04",
+            "os_version": "20.04",
+            "bits": 64,
+            "linux_distro": "Ubuntu",
+            "product": product,
+            "debug": False,
+            "browser_version": "98.0.2",
+            "browser_channel": channel,
+            "verify": False,
+            "headless": True,
+        }
+
+        result = {
+            "test": test_id,
+            "subtests": [
+                {
+                    "name": subtest_name,
+                    "status": subtest_status,
+                    "message": None,
+                    "known_intermittent": []
+                }
+            ],
+            "status": status,
+            "message": None,
+            "duration": 555,
+            "known_intermittent": []
+        }
+
+        if status != "OK":
+            result["expected"] = "OK"
+
+        if subtest_status != "PASS":
+            result["subtests"][0]["expected"] = "PASS"
+
+        data = {
+            "time_start": 1648629686379,
+            "run_info": run_info,
+            "results": [result],
+            "time_end": 1648629698721
+        }
+
+        path = os.path.join(tmp_path, f"wptreport-{len(created_files)}.json")
+        with open(path, "w") as f:
+            json.dump(data, f)
+
+        created_files.append(path)
+        return run_info, path
+
+    yield create_metadata
+
+    for path in created_files:
+        os.unlink(path)
+
+
+def test_update(tmp_path, metadata_file):
+    # This has to be a real test so it's in the manifest
```

Comment:
Will the test fail if that isn't the case? If it will spuriously pass, is there anything that can be asserted to avoid the risk of later rename?

Reply:
It will fail I think because we'll end up skipping the test we can't find in the manifest and so not generating metadata for it, so the later asserts will fail.

Reply:
Sounds right!